Practice/karazanov/lec_2.3.py

```diff
@@ -0,0 +1,83 @@
+# Спроектировать классы (один или несколько) для игры в танки и
+# создать экземпляры этих классов.
+class MediumTank:
+    def __init__(self, number, name, power, armour, speed):
+        self.number = number
+        self.name = name
+        self.power = power
+        self.armour = armour
+        self.speed = speed
+
+    def __dict__(self):
+        print(f"№{self.number}: {self.name}, {self.power=} {self.armour=} {self.speed=}")
+
+    def move(self):
+        new_position = distance + self.speed
+        print(f"Вы находитесь на {new_position} метров от стартовой точки")
+        return new_position
+
+    def shoot(self):
+        a = self.power
+        print(f"Вы нанесли {a} урона вражескому танку")
+        if enemy_tank == 1:
+            panther_tank.calculate_dmg()
+        elif enemy_tank == 2:
+            t34_tank.calculate_dmg()
+        return a
+
+    def calculate_dmg(self):
+        pass
+
+
+def pick_your_tank():
```

Comment:
Сюда (и в pick_enemy_tank, и в on_action_choice) можно передать список танков в качестве параметра (например, tank_lst или tanks). Тогда, если пользователь выберет номер танка (0 или 1), можно напечатать tanks[i].name. И это будет работать для любого количества любых танков. Сейчас, если добавится новый танк tiger, придется переписывать и pick_your_tank, и pick_enemy_tank, и on_action_choice, а это лишняя работа.

Reply:
Да, я тоже над этим задумывался, просто до решения не смог додуматься

Reply:
Исправил. Написал решение через:
```suggestion
def pick_your_tank(lst):
    tank_choice = int(input("Выберите номер своего танка (введите 1 или 2): "))
    print(f"Ваш танк - {lst[tank_choice].name}")
    return tank_choice


def pick_enemy_tank(lst):
    enemy_choice = int(input("Выберите номер танка врага (введите 1 или 2): "))
    print(f"Ваш танк - {lst[enemy_choice].name}")
    return enemy_choice
```

Reply:
ОК. Только в lst нумерация начинается с нуля. И эти изменения здесь пока не отобразились.

Reply:
Исправил. Залил новую версию